velox/common/memory/Memory.h

```diff
@@ -985,10 +875,9 @@ Allocator& MemoryManager<Allocator, ALIGNMENT>::getAllocator() {
 
 IMemoryManager& getProcessDefaultMemoryManager();
 
-// adds a new scoped child memory pool to the root
-// the new scoped child pool memory cap is set to the input value provided
-std::unique_ptr<ScopedMemoryPool> getDefaultScopedMemoryPool(
-    int64_t cap = kMaxMemory);
+/// Adds a new child memory pool to the root. The new child pool memory cap is
+/// set to the input value provided.
+std::shared_ptr<MemoryPool> getDefaultMemoryPool(int64_t cap = kMaxMemory);
```

Comment:
So this is not the "root" pool, but rather a new child? I wonder if there is a better name for that function.

Reply:
How about newMemoryPool and drop get and default. I'd leave this for a followup. Thanks for pointing out!